python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -1203,3 +1207,30 @@ def evaluate_for_asset(
                 context.asset_key, context.partitions_def, rate_limited_asset_partitions
             ),
         )
+
+
+@whitelist_for_serdes
+class SkipOnRunInProgressRule(AutoMaterializeRule, NamedTuple("_SkipOnRunInProgressRule", [])):
+    @property
+    def decision_type(self) -> AutoMaterializeDecisionType:
+        return AutoMaterializeDecisionType.SKIP
+
+    @property
+    def description(self) -> str:
+        return "in-progress asset"
+
+    def evaluate_for_asset(
+        self, context: AssetConditionEvaluationContext
+    ) -> "AssetConditionResult":
+        from .asset_condition.asset_condition import AssetConditionResult
+
+        planned_materialization_info = context.instance_queryer.instance.event_log_storage.get_latest_planned_materialization_info(
+            context.asset_key
+        )
+        if planned_materialization_info:
+            dagster_run = context.instance_queryer.instance.get_run_by_id(
+                planned_materialization_info.run_id
+            )
+            if dagster_run and dagster_run.status in IN_PROGRESS_RUN_STATUSES:
+                return AssetConditionResult.create(context, context.candidate_subset)
```

Comment:
Not absolutely vital for the initial PR, but we should include the run ID as metadata